conn/physic/units_test.go

```diff
@@ -745,6 +745,133 @@ func TestMaxInt64(t *testing.T) {
 	}
 }
 
+func TestSetHelper(t *testing.T) {
+	succeeds := []struct {
+		in        string
+		uintbase  prefix
+		expected  int64
+		usedChars int
+	}{
+		{"1p", pico, 1, 2},
+		{"1n", pico, 1000, 2},
+		{"1u", pico, 1000000, 2},
+		{"1µ", pico, 1000000, 3},
+		{"1m", pico, 1000000000, 2},
+		{"1k", pico, 1000000000000000, 2},
+		{"1M", pico, 1000000000000000000, 2},
+		{"1p", nano, 0, 2},
+		{"1n", nano, 1, 2},
+		{"1u", nano, 1000, 2},
+		{"1µ", nano, 1000, 3},
+		{"1m", nano, 1000000, 2},
+		{"1k", nano, 1000000000000, 2},
+		{"1M", nano, 1000000000000000, 2},
+		{"1G", nano, 1000000000000000000, 2},
+		{"1p", micro, 0, 2},
+		{"1n", micro, 0, 2},
+		{"1u", micro, 1, 2},
+		{"1µ", micro, 1, 3},
+		{"1m", micro, 1000, 2},
+		{"1k", micro, 1000000000, 2},
+		{"1M", micro, 1000000000000, 2},
+		{"1G", micro, 1000000000000000, 2},
+		{"1T", micro, 1000000000000000000, 2},
+	}
+
+	fails := []struct {
+		in     string
+		prefix prefix
+	}{
+		{"1G", pico},
+		{"1T", pico},
+		{"1T", nano},
+		{"not a number", nano},
+	}
+
+	for _, tt := range succeeds {
+		got, used, err := set(tt.in, tt.uintbase)
+
+		if got != tt.expected {
+			t.Errorf("set(%s,%d) wanted: %v(%d) but got: %v(%d)", tt.in, tt.uintbase, tt.expected, tt.expected, got, got)
+		}
+		if used != tt.usedChars {
+			t.Errorf("set(%s,%d) used %d chars but should used: %d chars", tt.in, tt.uintbase, used, tt.usedChars)
+		}
+		if err != nil {
+			t.Errorf("set(%s,%d) got unexpected error: %v", tt.in, tt.uintbase, err)
+		}
+	}
+
+	for _, tt := range fails {
+		_, _, err := set(tt.in, tt.prefix)
+
+		if err == nil {
+			t.Errorf("set(%s,%d) got expected error but got none", tt.in, tt.prefix)
```

Comment:
Nitpick: the first got shouldn't be there (set(%s,%d) <s>got</s> expected error but got none)

Reply:
I'll fix on with follow up.